internal/fflag/fflag.go

```diff
@@ -0,0 +1,133 @@
+package fflag
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	"github.com/superfly/flyctl/internal/tracing"
+)
+
+const clientSideID string = "6557a71bbffb5f134b84b15c"
+
+type FeatureFlagClient struct {
+	ldContext  ldcontext.Context
+	flags      map[string]FeatureFlag
+	flagsMutex sync.Mutex
+}
+
+type featureFlagClientContextKey struct{}
+
+func NewContextWithClient(ctx context.Context, ffClient *FeatureFlagClient) context.Context {
+	return context.WithValue(ctx, featureFlagClientContextKey{}, ffClient)
+}
+
+func ClientFromContext(ctx context.Context) *FeatureFlagClient {
+	client := ctx.Value(featureFlagClientContextKey{})
+	if client == nil {
+		return nil
+	}
+	return client.(*FeatureFlagClient)
+}
+
+type UserInfo struct {
+	OrganizationID string
+	UserID         int
+}
+
+func NewClient(ctx context.Context, userInfo UserInfo) (*FeatureFlagClient, error) {
+	_, span := tracing.GetTracer().Start(ctx, "new_feature_flag_client")
+	defer span.End()
+
+	orgID, err := strconv.Atoi(userInfo.OrganizationID)
+	if err != nil {
+		orgID = 0
+	}
+
+	orgContext := ldcontext.NewBuilder("flyctl").Anonymous(true).SetInt("id", orgID).Kind(ldcontext.Kind("organization")).Build()
+	userContext := ldcontext.NewBuilder("flyctl").Anonymous(true).SetInt("id", userInfo.UserID).Kind(ldcontext.Kind("user")).Build()
+
+	launchDarklyContext := ldcontext.NewMultiBuilder().Add(orgContext).Add(userContext).Build()
+
+	ffClient := &FeatureFlagClient{
+		ldContext:  launchDarklyContext,
+		flagsMutex: sync.Mutex{},
+	}
+
+	go func() {
+		for {
+			if ffClient == nil {
+				return
+			}
+
+			flags, err := ffClient.updateFeatureFlags()
+			if err != nil {
+				return
+			}
+
+			ffClient.flagsMutex.Lock()
+			ffClient.flags = flags
+			ffClient.flagsMutex.Unlock()
+
+			// the launchdarkly docs recommend polling every 30 seconds
+			time.Sleep(30 * time.Second)
+		}
+	}()
+
+	return ffClient, nil
+}
+
+func (ffClient *FeatureFlagClient) GetFeatureFlagValue(key string, defaultValue interface{}) interface{} {
+	ffClient.flagsMutex.Lock()
+	defer ffClient.flagsMutex.Unlock()
+
+	if flag, ok := ffClient.flags[key]; ok {
+		return flag.Value
+	}
+	return defaultValue
+
+}
+
+type FeatureFlag struct {
+	FlagVersion int         `json:"flagVersion"`
+	TrackEvents bool        `json:"trackEvents"`
+	Value       interface{} `json:"value"`
+	Version     int         `json:"version"`
+	Variation   int         `json:"variation"`
+}
```

Comment:
Minor nit. Go allows the use of `any` instead of `interface{}` which is a little easier to read.

Reply:
oh i had no idea, that's way nicer